pkg/openstack/openstackclient.go

```diff
@@ -0,0 +1,251 @@
+			Name:      "openstackclient",
+			Namespace: instance.Namespace,
+		},
+	}
+
+	helper.GetLogger().Info("Reconciling OpenStackClient", "OpenStackClient.Namespace", instance.Namespace, "OpenStackClient.Name", openstackclient.Name)
+	op, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), openstackclient, func() error {
+		openstackclient.Spec.ContainerImage = openstackClientImage
+
+		// the following are created/owned by keystoneclient
+		openstackclient.Spec.OpenStackConfigMap = "openstack-config"
+		openstackclient.Spec.OpenStackConfigSecret = "openstack-config-secret"
+
+		err := controllerutil.SetControllerReference(helper.GetBeforeObject(), openstackclient, helper.GetScheme())
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenStackControlPlaneClientReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			corev1beta1.OpenStackControlPlaneClientReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	if op != controllerutil.OperationResultNone {
+		helper.GetLogger().Info(fmt.Sprintf("OpenStackClient %s - %s", openstackclient.Name, op))
+	}
+
+	if openstackclient.Status.Conditions.IsTrue(condition.DeploymentReadyCondition) {
+		instance.Status.Conditions.MarkTrue(corev1beta1.OpenStackControlPlaneClientReadyCondition, corev1beta1.OpenStackControlPlaneClientReadyMessage)
+	} else {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenStackControlPlaneClientReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			corev1beta1.OpenStackControlPlaneClientReadyRunningMessage))
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// ClientPod func
+func ClientPod(
+	instance *corev1beta1.OpenStackClient,
+	labels map[string]string,
+	configHash string,
+	secretHash string,
+) *corev1.Pod {
+
+	clientPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      instance.Name,
+			Namespace: instance.Namespace,
+		},
+	}
+
+	var config0644AccessMode int32 = 0644
+	clientPod.Spec.Volumes = []corev1.Volume{
+		{
+			Name: "openstack-config",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: instance.Spec.OpenStackConfigMap,
+					},
+				},
+			},
+		},
+		{
+			Name: "openstack-config-secret",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: instance.Spec.OpenStackConfigSecret,
+				},
+			},
+		},
+		{
+			Name: "kolla-config",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					DefaultMode: &config0644AccessMode,
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "openstackclient-sh",
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "kolla_config.json",
+							Path: "config.json",
+						},
+					},
+				},
+			},
+		},
+	}
```

Comment:
Looks like this is redundant given line 197

Reply:
yes, I added a function for that. Removed in latest push.